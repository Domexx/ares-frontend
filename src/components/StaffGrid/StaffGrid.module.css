/* === === === === StaffGrid === === === === */

.StaffGrid {
  margin-top: 0;
}

.StaffGrid-wrapper {
  grid-column: 2 / span 12;
}

@media (--phone) {
  .StaffGrid-wrapper {
    grid-column: 2 / span 14;
  }
}
