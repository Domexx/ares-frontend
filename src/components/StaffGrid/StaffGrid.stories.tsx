import { Story } from '@storybook/react/types-6-0';
import React, { ComponentProps } from 'react';
import StaffGridComponent, { Props } from './StaffGrid';
import StaffGridMock from './StaffGrid.mock.json';

const Component = {
  title: 'Components/StaffGrid',
  component: StaffGridComponent,
  argTypes: {},
};

const Template: Story<ComponentProps<typeof StaffGridComponent>> = (args) => {
  return <StaffGridComponent {...args} />;
};

export const StaffGrid = Template.bind({});
export default Component;

const items = StaffGridMock;

// https://storybook.js.org/docs/react/essentials/controls
StaffGrid.args = {
  items,
} as Props;
