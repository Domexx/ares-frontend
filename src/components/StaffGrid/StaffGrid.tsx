import React from 'react';
import { StaffGridItem } from '../../shared/interfaces/StaffGridItem';
import List from './List';
import s from './StaffGrid.module.css';

export type Props = {
  items: StaffGridItem[];
};

const StaffGrid: React.FC<Props> = ({ items }) => {
  return (
    <section className={`${s['StaffGrid']} u-grid`}>
      <div className={s['StaffGrid-wrapper']}>
        <List items={items} />
      </div>
    </section>
  );
};

export default StaffGrid;
