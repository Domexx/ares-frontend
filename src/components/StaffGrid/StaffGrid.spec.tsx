import React from 'react';

import { render } from '@testing-library/react';

import StaffGrid from './StaffGrid';

describe('Staff Grid Component', () => {
  it('should render properly', () => {
    const { container } = render(
      <StaffGrid
        items={[
          {
            title: 'Administration',
            members: [
              {
                look: 'hd-180-1.ch-255-66.lg-280-110.sh-305-62.ha-1012-110.hr-828-61&action=std&gesture=std&direction=2&head_direction=2&size=l',
                motto: 'Time is just an illusion',
                name: 'Dome',
                online: false,
                rank: 'Developer',
              },
              {
                look: 'hd-180-1.ch-255-66.lg-280-110.sh-305-62.ha-1012-110.hr-828-61&action=std&gesture=std&direction=2&head_direction=2&size=l',
                motto: 'Time is just an illusion',
                name: 'Dome',
                online: true,
                rank: 'Developer',
              },
              {
                look: 'hd-180-1.ch-255-66.lg-280-110.sh-305-62.ha-1012-110.hr-828-61&action=std&gesture=std&direction=2&head_direction=2&size=l',
                motto: 'Time is just an illusion',
                name: 'Dome',
                online: false,
                rank: 'Developer',
              },
            ],
          },
          {
            title: 'Moderation',
            members: [
              {
                look: 'hd-180-1.ch-255-66.lg-280-110.sh-305-62.ha-1012-110.hr-828-61&action=std&gesture=std&direction=2&head_direction=2&size=l',
                motto: 'Time is just an illusion',
                name: 'Dome',
                online: false,
                rank: 'Developer',
              },
              {
                look: 'hd-180-1.ch-255-66.lg-280-110.sh-305-62.ha-1012-110.hr-828-61&action=std&gesture=std&direction=2&head_direction=2&size=l',
                motto: 'Time is just an illusion',
                name: 'Dome',
                online: false,
                rank: 'Developer',
              },
              {
                look: 'hd-180-1.ch-255-66.lg-280-110.sh-305-62.ha-1012-110.hr-828-61&action=std&gesture=std&direction=2&head_direction=2&size=l',
                motto: 'Time is just an illusion',
                name: 'Dome',
                online: true,
                rank: 'Developer',
              },
            ],
          },
        ]}
      />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
