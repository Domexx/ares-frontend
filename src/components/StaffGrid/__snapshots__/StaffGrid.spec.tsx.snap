// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Staff Grid Component should render properly 1`] = `
<section
  class="StaffGrid u-grid"
>
  <div
    class="StaffGrid-wrapper"
  />
</section>
`;
